src/api/ledger/pathfind.js

```diff
@@ -1,147 +1,99 @@
-const bignum = require('bignumber.js');
+const BigNumber = require('bignumber.js');
 const utils = require('./utils');
 const validate = utils.common.validate;
 const parsePathfind = require('./parse/pathfind');
 const NotFoundError = utils.common.errors.NotFoundError;
-const TimeOutError = utils.common.errors.TimeOutError;
+const composeAsync = utils.common.composeAsync;
 
-/**
- * Get a ripple path find, a.k.a. payment options,
- * for a given set of parameters and respond to the
- * client with an array of fully-formed Payments.
- *
- * @param {Remote} remote
- * @param {RippleAddress} req.params.source_account
- * @param {Amount Array ["USD r...,XRP,..."]} req.query.source_currencies
- *          - Note that Express.js middleware replaces "+" signs with spaces.
- *            Clients should use "+" signs but the values here will end up
- *            as spaces
- * @param {RippleAddress} req.params.destination_account
- * @param {Amount "1+USD+r..."} req.params.destination_amount_string
- */
-function getPathFind(pathfind, callback) {
-  const self = this;
-  validate.pathfind(pathfind);
+function addParams(params, result) {
+  return _.assign({}, result, {
+    source_account: params.src_account,
+    source_currencies: params.src_currencies,
+    destination_amount: params.dst_amount
+  });
+}
 
-  function prepareOptions() {
-    const pathfindParams = {
-      src_account: pathfind.source.address,
-      dst_account: pathfind.destination.address,
-      dst_amount: utils.common.toRippledAmount(pathfind.destination.amount)
-    };
-    if (typeof pathfindParams.dst_amount === 'object'
-          && !pathfindParams.dst_amount.issuer) {
-      // Convert blank issuer to sender's address
-      // (Ripple convention for 'any issuer')
-      // https://ripple.com/build/transactions/
-      //     #special-issuer-values-for-sendmax-and-amount
-      // https://ripple.com/build/ripple-rest/#counterparties-in-payments
-      pathfindParams.dst_amount.issuer = pathfindParams.dst_account;
-    }
-    if (pathfind.source.amounts && pathfind.source.amounts.length > 0) {
-      pathfindParams.src_currencies = pathfind.source.amounts;
-    }
-    return pathfindParams;
+function requestPathFind(remote, pathfind, callback) {
+  const params = {
+    src_account: pathfind.source.address,
+    dst_account: pathfind.destination.address,
+    dst_amount: utils.common.toRippledAmount(pathfind.destination.amount)
+  };
+  if (typeof params.dst_amount === 'object' && !params.dst_amount.issuer) {
+    // Convert blank issuer to sender's address
+    // (Ripple convention for 'any issuer')
+    // https://ripple.com/build/transactions/
+    //     #special-issuer-values-for-sendmax-and-amount
+    // https://ripple.com/build/ripple-rest/#counterparties-in-payments
+    params.dst_amount.issuer = params.dst_account;
+  }
+  if (pathfind.source.amounts && pathfind.source.amounts.length > 0) {
+    params.src_currencies = pathfind.source.amounts;
```

Comment:
`source.amounts` and `destination.amount` is most similar to the schema for `preparePayment`, which uses `source.amount` and `destination.amount`